src/hydra_endpoints_provider.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provider side of the hydra-endpoints-info relation interface."""
+import logging
+
+from ops.framework import Object
+
+logger = logging.getLogger(__name__)
+
+RELATION_NAME = "endpoint-info"
+
+
+class HydraEndpointsProvider(Object):
```

Comment:
If we are going to use a library then this class should be placed in the same library as the `HydraEndpointsRequirer`